src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -670,9 +673,14 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                         throw ExceptionUtilities.UnexpectedValue(_deconstruction.Kind());
                 }
 
-                TypeSymbol result = this._type;
-                Debug.Assert((object)result != null);
-                return result;
+                if ((object)this._type == null)
+                {
+                    diagnostics.Add(this.ForbiddenDiagnostic, Locations[0], Name);
+                    SetType(new ExtendedErrorTypeSymbol(
+                        ((SourceModuleSymbol)this.ContainingModule).DeclaringCompilation, name: "var", arity: 0, errorInfo: null, variableUsedBeforeDeclaration: true));
```

Comment:
>variableUsedBeforeDeclaration: true) [](start = 130, length = 36)

Why does it make sense to set ```variableUsedBeforeDeclaration``` to true here and what effect is this going to have?